.github/workflows/render-rmarkdown.yaml

```diff
@@ -0,0 +1,68 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
+# Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
+on:
+  push:
+    branches: [main, master]
+  pull_request:
+  release:
+    types: [published]
+  workflow_dispatch:
+
+name: render-rmarkdown.yaml
+
+permissions: read-all
+
+jobs:
+  render-rmarkdown:
+    runs-on: ubuntu-latest
+    # Only restrict concurrency for non-PR jobs
+    concurrency:
+      group: rmarkdown-${{ github.event_name != 'pull_request' || github.run_id }}
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    permissions:
+      contents: write
+      
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: 'latest'
+
+      - name: Setup Pandoc
+        uses: r-lib/actions/setup-pandoc@v2
+
+      - name: Setup R Dependencies
+        uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::rmarkdown
+
+      - name: Render RMarkdown files
+        shell: bash
+        run: |
+          RMD_PATH=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} | grep '\.Rmd$' || true)
+          if [ -n "$RMD_PATH" ]; then
+            Rscript -e 'for (f in commandArgs(TRUE)) if (file.exists(f)) rmarkdown::render(f)' ${RMD_PATH[*]}
+            echo "RMD_PATH=$RMD_PATH" >> $GITHUB_ENV
+          else
+            echo "No RMarkdown files changed."
+          fi
+
+      - name: Commit Results
+        shell: bash
+        run: |
+          if [ -n "$RMD_PATH" ]; then
+            MD_FILES=$(echo "$RMD_PATH" | sed 's/\.Rmd$/\.md/')
+            git config --local user.name "$GITHUB_ACTOR"
+            git config --local user.email "$GITHUB_ACTOR@users.noreply.github.com"
+            git add $MD_FILES
+            git commit -m 'Re-build RMarkdown files' || echo "No changes to commit"
+            git push origin || echo "No changes to commit"
+          else
+            echo "No Markdown files to commit."
+          fi
```

Comment:
Is this part of the action actually ever triggering? 

I would expect a commit with the message `Re-build RMarkdown files`, but I don't actually see one in the commit log for this PR.

Reply:
This, I didn't change that much from the [initial template](https://github.com/r-lib/actions/tree/v2/examples#render-rmarkdown). I'll look into this more.

Reply:
As I understand, we can make the commit in the current branch then merge or in `main` when we merge it. How it is set up now it would knit when merged with `main`. It is preferable to commit to current branch then merge into `main`. I'd thought about it but didn't make it happen.  I'll merge once I resolve this.

Reply:
See discussion below.